src/history/hs.c

```diff
@@ -261,6 +261,39 @@ __wt_hs_cursor_close(WT_SESSION_IMPL *session, uint32_t session_flags, bool is_o
     return (0);
 }
 
+/*
+ * __wt_hs_modify --
+ *     Make an update to the history store.
+ *
+ * History store updates don't use transactions as those updates should be immediately visible and
+ *     don't follow normal transaction semantics. For this reason, history store updates are
+ *     directly modified using the low level api instead of the ordinary cursor api.
+ */
+int
+__wt_hs_modify(WT_CURSOR_BTREE *hs_cbt, WT_UPDATE *hs_upd)
+{
+    WT_DECL_RET;
+    WT_PAGE_MODIFY *mod;
+    WT_SESSION_IMPL *session;
+    WT_UPDATE *last_upd;
+
+    session = (WT_SESSION_IMPL *)hs_cbt->iface.session;
+
+    /* Cope if we find existing updates. */
```

Comment:
Would mind asserting somewhere that there is no existing entry in the insert case?

Reply:
Sure, done in ef8bf5dab.